lib/fastlane/plugin/wpmreleasetoolkit/helper/metadata_download_helper.rb

```diff
@@ -4,12 +4,17 @@
 module Fastlane
   module Helper
     class MetadataDownloader
+      AUTO_RETRY_SLEEP_TIME = 20
+      MAX_AUTO_RETRY_ATTEMPTS = 30
+
       attr_reader :target_folder, :target_files
 
-      def initialize(target_folder, target_files)
+      def initialize(target_folder, target_files, auto_retry)
         @target_folder = target_folder
         @target_files = target_files
+        @auto_retry = auto_retry
         @alternates = {}
+        @auto_retry_attempt_counter = 0
```

Comment:
It just happened to me that [the `downloader` instance is reused at call site for each locale to be downloaded](https://github.com/wordpress-mobile/release-toolkit/pull/474/files#diff-1901c13107da3d09f269520d27db2427b6ceb5b09b76eaa9a443836a97c3fc85R19).

This means that if you only set the `@auto_retry_attempt_counter` in `initialize`, the counter will be shared across all usages of the downloader, and thus across all locales for which you call `download(…)` on.

Was this intentional to consider that `MAX_AUTO_RETRY_ATTEMPTS` be seen as "max 30 retries in total" as opposed to "max 30 retries per locale"? If not, I think we need to make sure `@auto_retry_attempt_counter` is reset to `0` every time the caller calls `download(…)` on a new locale.

For example, maybe you could move the current 3 LoC implementation of `download` to a private `try_download` method, then change `download(…)` implementation to now do `@auto_retry_attempt_counter = 0` + call `try_download(…)`?

Reply:
SGTM 👍